homeassistant/components/nws/weather.py

```diff
@@ -0,0 +1,378 @@
+                "Freezing rain",
+                "Rain/freezing rain",
+            ],
+        ),
+        ("hail", []),
+        (
+            "lightning-rainy",
+            [
+                "Thunderstorm (high cloud cover)",
+                "Thunderstorm (medium cloud cover)",
+                "Thunderstorm (low cloud cover)",
+            ],
+        ),
+        ("lightning", []),
+        ("pouring", []),
+        (
+            "rainy",
+            [
+                "Rain",
+                "Rain showers (high cloud cover)",
+                "Rain showers (low cloud cover)",
+            ],
+        ),
+        ("windy-variant", ["Mostly cloudy and windy", "Overcast and windy"]),
+        (
+            "windy",
+            [
+                "Fair/clear and windy",
+                "A few clouds and windy",
+                "Partly cloudy and windy",
+            ],
+        ),
+        ("fog", ["Fog/mist"]),
+        ("clear", ["Fair/clear"]),  # sunny and clear-night
+        ("cloudy", ["Mostly cloudy", "Overcast"]),
+        ("partlycloudy", ["A few clouds", "Partly cloudy"]),
+    ]
+)
+
+ERRORS = (aiohttp.ClientError, JSONDecodeError)
+
+FORECAST_MODE = ["daynight", "hourly"]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Optional(CONF_LATITUDE): cv.latitude,
+        vol.Optional(CONF_LONGITUDE): cv.longitude,
+        vol.Optional(CONF_MODE, default="daynight"): vol.In(FORECAST_MODE),
+        vol.Optional(CONF_STATION): cv.string,
+        vol.Required(CONF_API_KEY): cv.string,
+    }
+)
+
+
+def convert_condition(time, weather):
+    """
+    Convert NWS codes to HA condition.
+
+    Choose first condition in CONDITION_CLASSES that exists in weather code.
+    If no match is found, return first condition from NWS
+    """
+    conditions = [w[0] for w in weather]
+    prec_probs = [w[1] or 0 for w in weather]
+
+    # Choose condition with highest priority.
+    cond = next(
+        (
+            key
+            for key, value in CONDITION_CLASSES.items()
+            if any(condition in value for condition in conditions)
+        ),
+        conditions[0],
+    )
+
+    if cond == "clear":
+        if time == "day":
+            return "sunny", max(prec_probs)
+        if time == "night":
+            return "clear-night", max(prec_probs)
+    return cond, max(prec_probs)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the NWS weather platform."""
+    from pynws import SimpleNWS
+
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+    station = config.get(CONF_STATION)
+    api_key = config[CONF_API_KEY]
+    mode = config[CONF_MODE]
+
+    if None in (latitude, longitude):
+        _LOGGER.error("Latitude/longitude not set in Home Assistant config")
+        return ConfigEntryNotReady
```

Comment:
If it's a non recoverable error we should just `return` here after logging the error. We can't raise this error since this isn't a config entry.